src/addons/transitions/ReactCSSTransitionGroupChild.js

```diff
@@ -85,7 +87,11 @@ var ReactCSSTransitionGroupChild = React.createClass({
     CSSCore.addClass(node, className);
 
     // Need to do this to actually trigger a transition.
-    this.queueClass(activeClassName);
+    if ( REQUEST_ANIMATION_FRAME ) {
+      REQUEST_ANIMATION_FRAME(this.queueClass.bind(this, activeClassName));
```

Comment:
`queueClass` calls `setTimeout` (I think in an attempt to make sure we group changes together) so now we're asyncing twice in a row with a non-trivial chance that this change is not batched with the previous one (which might actually be the problem). `TICK` is set to 17ms which is almost like "do right after next animation frame".

It might be that we just want to use `requestAnimationFrame` in `queueClass` (and do the same timeoutid tracking/clearing).

I don't really know though, I haven't written any of this. It just feels weird to requestAnimationFrame -> setTimeout, feels like too many of the hacks done to get around mysterious issues with async code.